ui/admin/app/components/form/target/details/index.js

```diff
@@ -49,28 +51,75 @@ export default class FormTargetComponent extends Component {
    * @type {boolean}
    */
   get showUpdateWorkerFilterButton() {
-    return this.args.model.worker_filter?.length;
+    return this.args.model.worker_filter;
   }
 
-  get showEgressFilter() {
-    return !this.args.model.worker_filter?.length && !this.args.model.isNew;
+  /**
+   * determines when the filters should be shown
+   * @type {boolean}
+   */
+  get showTargetFilters() {
+    return !this.args.model.worker_filter;
+  }
+
+  /**
+   * determines when the enterprise dual filter diagrams should be shown
+   * @type {boolean}
+   */
+  get workerDiagramDualFilter() {
+    return this.features.isEnabled('target-worker-filters-v2-ingress');
+  }
+  /**
+   * determines when the hcp dual filter diagrams should be shown
+   * @type {boolean}
+   */
+  get workerDiagramHCPDualFilter() {
+    return this.features.isEnabled('target-worker-filters-v2-hcp');
+  }
+
+  /**
+   * determines when the dual filter fields should be shown
+   * @type {boolean}
+   */
+  get showDualTargetWorkerFilters() {
+    return this.workerDiagramHCPDualFilter || this.workerDiagramDualFilter;
+  }
+
+  /**
+   * determines when the single filter should be shown
+   * @type {boolean}
+   */
+  get showOSSTargetWorkerFilters() {
+    return (
+      this.features.isEnabled('target-worker-filters-v2') &&
+      workerFilters.filter((type) => type !== 'ingress_worker_filter')
+    );
   }
 
   //actions
   @action
-  toggleEgressWorkerFilter() {
-    this.egressWorkerFilterEnabled = !this.egressWorkerFilterEnabled;
+  toggleFilter(filter) {
+    if (filter === 'egress_worker_filter') {
+      this.egressWorkerFilterEnabled = !this.egressWorkerFilterEnabled;
+    } else {
+      this.ingressWorkerFilterEnabled = !this.ingressWorkerFilterEnabled;
+    }
+
     if (!this.egressWorkerFilterEnabled) {
       this.args.model.egress_worker_filter = '';
     }
+    if (!this.ingressWorkerFilterEnabled) {
+      this.args.model.ingress_worker_filter = '';
+    }
   }
```

Comment:
The branching makes this, in essence, two separate functions.  Ideally it would be generic.  Is there a way to make it so?  If not, can it be split into separate actions?